File: backend/src/BiteRight.Application/Dtos/Products/ExpirationDateKindDto.cs
// # ==============================================================================
// # Solution: BiteRight
// # File: ExpirationDateKindDto.cs
// # Author: Łukasz Sobczak
// # Created: 11-02-2024
// # ==============================================================================

namespace BiteRight.Application.Dtos.Products;

public enum ExpirationDateKindDto
{
    Unknown = 0,
    Infinite = 1,
    BestBefore = 2,
    UseBy = 3
}

File: mobile/app/src/main/java/com/sobczal2/biteright/dto/products/ExpirationDateKindDto.kt
package com.sobczal2.biteright.dto.products

import com.sobczal2.biteright.R

enum class ExpirationDateKindDto(val value: Int) {
    Unknown(0),
    Infinite(1),
    BestBefore(2),
    UseBy(3);

    fun toLocalizedResourceID() = when (this) {
        Unknown -> R.string.unknown
        Infinite -> R.string.infinite
        BestBefore -> R.string.best_before
        UseBy -> R.string.use_by
    }

    fun shouldIncludeDate() = when (this) {
        Unknown -> false
        Infinite -> false
        BestBefore -> true
        UseBy -> true
    }
}